src/super_gradients/training/datasets/depth_estimation_datasets/redweb_dataset.py

```diff
@@ -0,0 +1,57 @@
+import os
+
+from typing import List
+
+from PIL import Image
+
+from super_gradients.training.datasets.depth_estimation_datasets import AbstractDepthEstimationDataset
+from super_gradients.training.transforms.depth_estimation import AbstractDepthEstimationTransform
+from super_gradients.training.samples import DepthEstimationSample
+from super_gradients.common.object_names import Datasets
+from super_gradients.common.registry.registry import register_dataset
+from super_gradients.common.decorators.factory_decorator import resolve_param
+from super_gradients.common.factories.transforms_factory import TransformsFactory
+
+
+@register_dataset(Datasets.REDWEB_DEPTH_ESTIMATION_DATASET)
+class ReDWebDepthEstimationDataset(AbstractDepthEstimationDataset):
+    """
+    Dataset class for training depth estimation models using ReDWeb V1 dataset.
+
+    Link to download dataset: https://sites.google.com/site/redwebcvpr18/
+    Paper: https://openaccess.thecvf.com/content_cvpr_2018/papers/Xian_Monocular_Relative_Depth_CVPR_2018_paper.pdf
+
+    The ReDWeb V1 dataset consists of 3.6K RGB-RD images, covering both indoor and outdoor scenes.
+    """
+
+    @resolve_param("transforms", factory=TransformsFactory())
+    def __init__(
+        self,
+        data_dir: str,
+        transforms: List[AbstractDepthEstimationTransform] = [],
+        images_dir: str = "Imgs",
+        targets_dir: str = "RDs",
+        image_extension: str = "jpg",
+        target_extension: str = "png",
+    ):
+        """ """
+        super().__init__(transforms=transforms)
+        self.data_dir = data_dir
+        self.images_dir = images_dir
+        self.targets_dir = targets_dir
+        self.image_extension = image_extension
+        self.target_extension = target_extension
+
+        self.pair_names = [name.split(".")[0] for name in os.listdir(os.path.join(self.data_dir, self.targets_dir))]
+
+    def __len__(self):
+        return len(self.pair_names)
+
+    def load_sample(self, index: int) -> DepthEstimationSample:
+        pair_name = self.pair_names[index]
+
+        image = Image.open(os.path.join(self.data_dir, self.images_dir, f"{pair_name}.{self.image_extension}")).convert("RGB")
+
+        depth_map = Image.open(os.path.join(self.data_dir, self.targets_dir, f"{pair_name}.{self.target_extension}"))
```

Comment:
I think somewhere here we have to have the option to pass (or at least know) how the "invalid" areas are marked in the sample.
Invalid areas are the ones that are missing depth information. I imagine this is crucial information should not be lost as transformations should respect that and do not change those areas as well as metrics which should respect that.
So I have two ideas about this:
- Either introduce a convention that all non-finite values in mask are treated as ignored/invalid.
- Or introduce a field in DepthEstimationSample (`ignore_index` or something) that would have the value corresponding to ignored areas.